CMakeLists.txt

```diff
@@ -0,0 +1,201 @@
+cmake_minimum_required(VERSION 3.9 FATAL_ERROR)
+project(rubberband VERSION 2.1.1)
+
+option(BUILD_EXECUTABLE "Build the Rubberband executable" OFF)
+option(BUILD_JNI "Build the Java Native Interface (JNI)" OFF)
+option(OPTIMIZE "Build Rubberband optimized for the build machine's CPU" OFF)
+option(USE_FFTW "Use FFTW instead of kissFFT" OFF)
+option(USE_LIBSAMPLERATE "Use libsamplerate instead of Speex" ON)
+
+add_library(rubberband
+  src/RubberBandStretcher.cpp
+  src/StretchCalculator.cpp
+  src/StretcherChannelData.cpp
+  src/StretcherImpl.cpp
+  src/StretcherProcess.cpp
+  src/audiocurves/CompoundAudioCurve.cpp
+  src/audiocurves/ConstantAudioCurve.cpp
+  src/audiocurves/HighFrequencyAudioCurve.cpp
+  src/audiocurves/PercussiveAudioCurve.cpp
+  src/audiocurves/SilentAudioCurve.cpp
+  src/audiocurves/SpectralDifferenceAudioCurve.cpp
+  src/base/Profiler.cpp
+  src/dsp/AudioCurveCalculator.cpp
+  src/dsp/FFT.cpp
+  src/dsp/Resampler.cpp
+  src/rubberband-c.cpp
+  src/system/Allocators.cpp
+  src/system/Thread.cpp
+  src/system/VectorOpsComplex.cpp
+  src/system/sysutils.cpp
+)
+
+if(BUILD_JNI)
+  add_library(rubberband-jni src/jni/RubberBandStretcherJNI.cpp)
+  target_include_directories(rubberband-jni PRIVATE src)
+
+  find_package(JNI REQUIRED)
+  target_include_directories(rubberband-jni PRIVATE ${JNI_INCLUDE_DIRS})
+  target_link_libraries(rubberband-jni PRIVATE ${JNI_LIBRARIES})
+endif()
+
+
+#
+# Dependencies
+#
+
+# Threads
+find_package(Threads REQUIRED)
+target_link_libraries(rubberband PRIVATE Threads::Threads)
+
+# FFTW / KissFFT
+if(USE_FFTW)
+  message(STATUS "Using FFT library: fftw")
+  find_path(FFTW3_INCLUDEDIR fftw3.h REQUIRED)
+  find_library(FFTW3_LIBRARY fftw3 REQUIRED)
+  target_include_directories(rubberband PRIVATE ${FFTW3_INCLUDEDIR})
+  target_link_libraries(rubberband PUBLIC ${FFTW3_LIBRARY})
+  target_compile_definitions(rubberband PRIVATE HAVE_FFTW3 FFTW_DOUBLE_ONLY)
+else()
+  message(STATUS "Using resampling library: kissfft")
+  target_compile_definitions(rubberband PRIVATE USE_KISSFFT)
+  target_sources(rubberband PRIVATE src/kissfft/kiss_fft.c src/kissfft/kiss_fftr.c)
+endif()
+
+# libsamplerate / Speex
+if(USE_LIBSAMPLERATE)
+  message(STATUS "Using resampling library: libsamplerate")
+  find_path(SAMPLERATE_INCLUDEDIR samplerate.h REQUIRED)
+  find_library(SAMPLERATE_LIBRARY samplerate REQUIRED)
+  target_include_directories(rubberband PRIVATE ${SAMPLERATE_INCLUDEDIR})
+  target_link_libraries(rubberband PUBLIC ${SAMPLERATE_LIBRARY})
+  target_compile_definitions(rubberband PRIVATE HAVE_LIBSAMPLERATE)
```

Comment:
libsamplerate just made a new release a few hours ago that adds CMake config files

Reply:
But libsamplerate still has autotools support too so many distributions will probably not switch to CMake unless that is removed...